mlflow/tracking/_tracking_service/client.py

```diff
@@ -198,6 +204,30 @@ def get_trace_info(self, request_id):
         """
         return self.store.get_trace_info(request_id)
 
+    def search_traces(
+        self,
+        experiment_ids: List[str],
+        filter_string: Optional[str] = None,
+        max_results: int = SEARCH_TRACES_MAX_RESULTS,
+        order_by: Optional[List[str]] = None,
+        page_token: Optional[str] = None,
+    ):
+        trace_infos, token = self.store.search_traces(
+            experiment_ids=experiment_ids,
+            filter_string=filter_string,
+            max_results=max_results,
+            order_by=order_by,
+            page_token=page_token,
+        )
+        traces = [
+            Trace(
+                trace_info=trace_info,
+                trace_data=self._download_trace_data(trace_info.request_id),
+            )
```

Comment:
This can be parallelized.

Reply:
Can we implement that?

Reply:
can we implement this as a follow-up?

Reply:
What's the performance look like currently? If it's really slow, we should do it now

Reply:
```
res = client.search_traces(experiment_ids=[exp.experiment_id], max_results=50)
```

took 45 seconds on my machine. Implementing parallel downloads now.